### PR TITLE
DSD-1450: Storybook 7 Radio & RadioGroup

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1450](https://jira.nypl.org/browse/DSD-1450)

## This PR does the following:

- Updates the Storybook documentation to version 7 for the `Radio` and `RadioGroup` components.
- Compare with `Radio` prod: https://nypl.github.io/nypl-design-system/reservoir/v1/?path=/docs/components-form-elements-radio--radio
- Compare with `RadioGroup` prod: https://nypl.github.io/nypl-design-system/reservoir/v1/?path=/docs/components-form-elements-radiogroup--radio-group

## How has this been tested?

Locally.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/Radio/Radio.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+- [W3C Radio Button Example](https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/radio/radio-1/radio-1.html)
+- [MDN input type="radio"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio)
+- [Deque University Radio and Radio Group](https://dequeuniversity.com/library/aria/radio-and-radio-group)
+- [Chakra UI Radio](https://chakra-ui.com/docs/components/form/radio)
+
+## Browser States
+
+<Canvas of={RadioStories.BrowserStates} />
+
+## With Text
```
Comment (bigfishdesign13) [inline src/components/Radio/Radio.mdx]: Maybe `Helper and Error Text`?

File: src/components/Radio/Radio.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+<Canvas of={RadioStories.WithText} />
+
+## With JSX Element Label
```
Comment (bigfishdesign13) [inline src/components/Radio/Radio.mdx]: Remove `with`.

File: src/components/Radio/Radio.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+};
+
+export default meta;
+type Story = StoryObj<typeof Radio>;
+
+/**
+ * Main Story for the Radio component. This must contains the `args`
+ * and `parameters` properties in this object.
+ */
+export const Controls: Story = {
```
Comment (bigfishdesign13) [inline src/components/Radio/Radio.stories.tsx:44]: To match the other stories, change to `WithControls`.

File: src/components/RadioGroup/RadioGroup.mdx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+Use the `layout` prop to set the `Radio` buttons to display in a column or in
+a row.
+
+<Canvas of={RadioGroupStories.LayoutPatterns} />
+
+## Browser States
+
+<Canvas of={RadioGroupStories.BrowserStates} />
+
+## With JSX Element labels
```
Comment (bigfishdesign13) [inline src/components/RadioGroup/RadioGroup.mdx:57]: Remove `With` and capitalize `Labels`.

File: src/components/Radio/Radio.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+            isChecked
+            labelText="I am checked and disabled"
+          />
+        </HStack>
+      </Box>
+    </VStack>
+  ),
+};
+
+export const WithText: Story = {
```
Comment (bigfishdesign13) [inline src/components/Radio/Radio.stories.tsx]: Add `name: "With Helper and Error Text"`.

File: src/components/Radio/Radio.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+export const JSXElement: Story = {
+  render: () => (
+    <Radio
+      id="jsx-label"
+      labelText={<span>Arts</span>}
+      name="jsxElementLabel"
+    />
+  ),
+  name: "With JSX Element Label",
```
Comment (bigfishdesign13) [inline src/components/Radio/Radio.stories.tsx]: Remove `With`.

File: src/components/Radio/Radio.mdx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+| Added             | `0.22.0`   |
+| Latest            | `1.5.0`    |
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Component Props](#component-props)
+- [Accessibility](#accessibility)
+- [Browser States](#browser-states)
+- [With Text](#with-text)
```
Comment (bigfishdesign13) [inline src/components/Radio/Radio.mdx]: Update TOC to match suggested section names (see comments).

File: src/components/RadioGroup/RadioGroup.mdx [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+| Latest            | `1.5.0`    |
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Component Props](#component-props)
+- [Accessibility](#accessibility)
+- [Layout Patterns](#layout-patterns)
+- [Browser States](#browser-states)
+- [With JSX Element labels](#with-jsx-element-labels)
```
Comment (bigfishdesign13) [inline src/components/RadioGroup/RadioGroup.mdx:21]: Remove `With` and capitalize `Labels`.

File: src/components/RadioGroup/RadioGroup.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+            <span>Math</span>
+            <Spacer />
+            <span>3</span>
+          </Flex>
+        }
+        value="Math"
+      />
+    </RadioGroup>
+  ),
+  name: "With JSX Element Label",
```
Comment (bigfishdesign13) [inline src/components/RadioGroup/RadioGroup.stories.tsx:234]: Remove `With` and pluralize `Labels`.

File: src/components/RadioGroup/RadioGroup.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+  render: () => <RadioGroupControlledExample />,
+};
+
+function RadioGroupUncontrolledExample() {
+  const radioGroupRef = createRef<HTMLDivElement>();
+  // Get the value through:
+  const onSubmit = (e) => {
+    e.preventDefault();
+    const radioGroupValue = (radioGroupRef.current as any).value;
+    console.log(`Uncontrolled example. Selected: ${radioGroupValue}`);
```
Comment (bigfishdesign13) [inline src/components/RadioGroup/RadioGroup.stories.tsx:266]: `radioGroupValue` is returning `undefined`.

Comment (bigfishdesign13) [reply src/components/RadioGroup/RadioGroup.stories.tsx:266]: Oops, I forgot this is a known issue.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Some minor wording changes.

File: src/components/Radio/Radio.stories.tsx (hunk 3 above)
Comment (EdwinGuzman) [reply src/components/Radio/Radio.stories.tsx:44]: Wait, I thought we said we would remove the word "With" from now on?

Comment (bigfishdesign13) [reply src/components/Radio/Radio.stories.tsx:44]: I guess we shgould talk about this a bit more.  In my head, I imagined we would keep "With" on the first story, but remove it from the others.

Leave it as is for now, but let's talk about this and make a final decision.  Then we can do one final sweep to clean up the story names.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Pull request merged**